File: app/models/patient/Patient.scala
package models.patient

import models.fhir.XmlRepresentation
import models.person.{HumanName, Person}

case class Patient(
  id: String,
  identifier: Seq[String],
  name: HumanName,
  gender: String
) extends Person with XmlRepresentation {

  lazy val toXML = XmlWrites.patientWrites(this)

}

File: frontend/src/models/patient/patient.ts
import {HumanName} from "../person/human-name";

export class Patient{
    
    id: string;
    identifier: Array<string>;
    name: HumanName;
    gender: string;

    constructor(data: any){
        Object.assign(this, data);
        this.name = new HumanName(data.name);
    }

    get number(){
        return this.identifier[0];
    }

}